components/calendar/src/types.rs

```diff
@@ -92,6 +93,35 @@ impl FormattableYear {
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
 pub struct MonthCode(pub TinyStr4);
 
+impl MonthCode {
+    /// Returns an option which is Some containing the non-month version of a leap month
+    /// if the MonthCode this method is called upon is a leap month, and None otherwise.
+    /// This method assumes the MonthCode is valid.
+    pub fn get_normal_if_leap(self) -> Option<MonthCode> {
+        let bytes = self.0.all_bytes();
+        if bytes[3] == b'L' {
+            Some(MonthCode(TinyAsciiStr::from_bytes(&bytes[0..3]).ok()?))
+        } else {
+            None
+        }
```

Comment:
Question: how does this code not create a `clippy::indexing_slicing` failure? Is Clippy smart enough to know that this is a `[u8; 4]` and that the indices are in range?

Reply:
Yes